File: src/mengka/trait_02/Taa.scala
package mengka.trait_02

/**
 * 同时继承多个Trait
 *
 * User: mengka.hyy
 * Date: 14-6-15-下午4:08
 */
object Taa {

  def main(args: Array[String]): Unit = {

    def mengka = new Mengka()

    pbaicai(mengka)
    pqingcai(mengka)

    mengka.planted()

  }

  def pbaicai(baicai: Baicai) {
    println(baicai.bname)
    baicai.planted()
  }

  def pqingcai(qingcai: Qingcai) {
    println(qingcai.qname)
    qingcai.planted()
  }

}

File: src/mengka/followObject_java_01/Taa.java
package mengka.followObject_java_01;

/**
 * 
 *  调用伴生对象的方法的时候，使用：Baicai$.MODULE$.xxx();
 * 
 * @author hyy044101331
 *
 */
public class Taa{
	
	public static void main(String[] args){
		
		/**
		 *  java调用伴生类方法
		 */
		new Mengka().planted();
		
	    /**
	     * java调用伴生对象方法
	     */
		Mengka$.MODULE$.planted();
	}
}